simulator/modules/execute/execute.cpp

```diff
@@ -104,7 +104,11 @@ void Execute<FuncInstr>::clock( Cycle cycle)
         if ( bypass_source.command_port->is_ready( cycle))
         {
             const auto bypass_direction = bypass_source.command_port->read( cycle).get_bypass_direction();
-            const auto data = bypass_source.data_ports.at( bypass_direction)->read( cycle).first;
+            auto data = bypass_source.data_ports.at( bypass_direction)->read( cycle).first;
```

Comment:
This line seems to be redundant now

Reply:
I catch unused variable (data) warning when I delete this line.

Reply:
Why not have something like:
```c++
auto& port = bypass_source.data_ports.at( bypass_direction);
FuncInstr::RegisterUInt data{};
while ( port->is_ready( cycle))
    data = port->read( cycle).first;
instr.set_v_src( data, src_index);
```